synse/http_integration_test.go

```diff
@@ -603,18 +614,38 @@ func TestIntegration_TagsOptions(t *testing.T) {
 			},
 		},
 		{
-			"multiple tags, multiple matches",
+			"multiple tag groups, valid tags, multiple matches",
 			scheme.ReadOptions{
-				Tags: []string{"foo/bar", "system/type:temperature"},
+				Tags: []string{"system/type:led", "system/type:temperature"},
 			},
 			map[string]int{
-				"total":       2,
-				"led":         0,
+				"total":       5,
+				"led":         2,
+				"temperature": 3,
+			},
+		},
+		{
+			"multiple tag groups, valid tags, multiple matches (shared)",
+			scheme.ReadOptions{
+				Tags: []string{"system/type:led", "foo/bar"},
+			},
+			map[string]int{
+				"total":       4,
+				"led":         2,
 				"temperature": 2,
 			},
 		},
-
-		// TODO - refer to #26: update tests for multiple tag group query.
+		{
+			"multiple tag groups, invalid tags, no match",
+			scheme.ReadOptions{
+				Tags: []string{"bar/foo/1", "bar/foo/2"},
```

Comment:
here, "invalid tag" is correct, because `bar/foo/2` is not a valid tag structure. (`bar/foo:2` would be)